L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc

```diff
@@ -33,18 +33,38 @@ void TTStubBuilder< Ref_Phase2TrackerDigi_ >::produce( edm::Event& iEvent, const
   edm::Handle< edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > > > clusterHandle;
   iEvent.getByToken( clustersToken, clusterHandle );
 
-  /// Get the maximum number of stubs per ROC
-  /// (CBC3-style)
-  //  unsigned maxStubs = theStackedTracker->getCBC3MaxStubs();  
-  unsigned maxStubs = 0; // 0 is default, cut is disabled
-
-  for (auto gd=theTrackerGeom->dets().begin(); gd != theTrackerGeom->dets().end(); gd++) {
-      DetId detid = (*gd)->geographicalId();
-      if(detid.subdetId()!=StripSubdetector::TOB && detid.subdetId()!=StripSubdetector::TID ) continue; // only run on OT
-      if(!tTopo->isLower(detid) ) continue; // loop on the stacks: choose the lower arbitrarily
-      DetId lowerDetid = detid;
-      DetId upperDetid = tTopo->partnerDetId(detid);
-      DetId stackDetid = tTopo->stack(detid);
+  int nmod=-1;
+
+  // Loop over all the tracker elements
+
+  for (auto gd=theTrackerGeom->dets().begin(); gd != theTrackerGeom->dets().end(); gd++)
+  {
+    DetId detid = (*gd)->geographicalId();
+    if(detid.subdetId()!=StripSubdetector::TOB && detid.subdetId()!=StripSubdetector::TID ) continue; // only run on OT
+    if(!tTopo->isLower(detid) ) continue; // loop on the stacks: choose the lower arbitrarily
+    DetId lowerDetid = detid;
+    DetId upperDetid = tTopo->partnerDetId(detid);
+    DetId stackDetid = tTopo->stack(detid);
+    bool isPS        = (theTrackerGeom->getDetectorType(stackDetid)==TrackerGeometry::ModuleType::Ph2PSP);
+
+    bool is10G_PS = false;
+    
+    // Determine if this module is a 10G transmission scheme module
+
+    if (detid.subdetId()==StripSubdetector::TOB)
+    {
+      if (tTopo->layer(detid)==1) is10G_PS = true;
+    }
+    else if (detid.subdetId()==StripSubdetector::TID)
+    {
+      if (tTopo->tidWheel(detid)<=2 && tTopo->tidRing(detid)<=4) is10G_PS = true;
```

Comment:
Can these magic numbers be obtained from some geometry/topology class? Otherwise, should be made into descriptive constants

Reply:
Done, numbers have been passed as joboptions